src/CMakeLists.txt

```diff
@@ -0,0 +1,84 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2013-2022 The Foundry Visionmongers Ltd
+
+
+#----------------------------------------------------------------------
+# Public headers
+
+set(public_header_source_root ${CMAKE_CURRENT_LIST_DIR}/include)
+
+# Installation location for install phase.
+install(
+    DIRECTORY
+    ${public_header_source_root}/openassetio
+    DESTINATION
+    ${CMAKE_INSTALL_INCLUDEDIR}
+)
+
+
+#-----------------------------------------------------------------------
+# Create core target
+
+# Note: static vs. shared is auto-determined by CMake's built-in
+# BUILD_SHARED_LIBS option.
+# TODO(DF): Allow customising library name (e.g. suffix)? See OCIO.
+add_library(openassetio)
+# Set good default target options.
+include(DefaultTargetProperties)
+set_default_target_properties(openassetio)
+# Add to the set of installable targets.
+install(TARGETS openassetio EXPORT ${PROJECT_NAME}_EXPORTED_TARGETS)
+
+# TODO(DF): Generate and install .pdb on Windows. See OCIO.
+
+
+#-----------------------------------------------------------------------
+# Target dependencies
+
+# Source file dependencies.
+target_sources(openassetio
+    PRIVATE
+    openassetio/managerAPI/ManagerInterface.cpp)
+
+# Public header dependency.
+target_include_directories(openassetio
+    PUBLIC
+    # Use includes from source tree for building.
+    "$<BUILD_INTERFACE:${public_header_source_root}>"
+    # Use includes from install tree for installed lib.
+    # TODO(DF): Not super sure about this - needs a test!
```

Comment:
Check the generated targets config file, and as long as the include directory specifier on the imported target is explicitly relative to the prefix variable, then it's all good.

Reply:
Hmm, I think this means it's good. Given an install directory structure of
```
.
├── include
│   └── openassetio
│       ├── export.h
│       └── managerAPI
│           └── ManagerInterface.hpp
└── lib
    ├── cmake
    │   └── OpenAssetIO
    │       ├── OpenAssetIOConfig.cmake
    │       ├── OpenAssetIOConfigVersion.cmake
    │       ├── OpenAssetIOTargets.cmake
    │       └── OpenAssetIOTargets-debug.cmake
    ├── libopenassetio.so -> libopenassetio.so.0.0
    ├── libopenassetio.so.0.0 -> libopenassetio.so.0.0.0
    └── libopenassetio.so.0.0.0
```
then in `OpenAssetIOTargets.cmake`
```
get_filename_component(_IMPORT_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)
get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)

<snip>

set_target_properties(OpenAssetIO::openassetio PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
)
```
At first I thought the multiple `get_filename_component` was a bug, but it must be just a way to (repeatedly) get the parent path.

Reply:
Yeah that's exactly how get_filename_component with `PATH` is meant to work.

All looks fine to me. Nice and relocatable.